verifier/src/lib.rs

```diff
@@ -51,114 +60,122 @@ impl<E: DisplayableError> VerificationResult<E> {
     /// representation with.
     pub fn fmt_padded(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result {
         if self.is_ok.into() {
-            write!(f, "{:pad$}Passed", "")?;
+            self.metadata.fmt_success(f, pad)?;
         } else {
-            self.error.fmt_padded(f, pad)?;
+            self.metadata.fmt_failure(f, pad)?;
         }
         Ok(())
     }
 }
 
-/// Number of spaces to indent nested error messages.
-const ERROR_INDENT: usize = 2;
+impl<T: ResultMessage> Display for VerificationResult<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        self.fmt_padded(f, 0)
+    }
+}
 
-/// An error that implements the [`Display`] trait.
-pub trait DisplayableError: Display + Clone {
-    /// Format the error with preceding padding
+/// A message that can be used in [`VerificationResult`]'s [`Display`]
+/// implementation.
+pub trait ResultMessage {
+    /// Formatting of a successful verification with this message type
     ///
-    /// The `pad` is the number of spaces to precede each line of the error
+    /// The `pad` is the number of spaces to precede each line of the message
     /// with.
-    fn fmt_padded(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result {
-        write!(f, "{:pad$}{self}", "")
-    }
+    fn fmt_success(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result;
+
+    /// Formatting of a failure verification with this message type
+    ///
+    /// The `pad` is the number of spaces to precede each line of the message
+    /// with.
+    fn fmt_failure(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result;
 }
 
 /// Failed to verify.
-#[derive(displaydoc::Display, Debug, Eq, PartialEq, Clone)]
-pub enum VerificationError {
-    /// A general error.
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub enum VerificationResultMetadata {
+    /// Generic no extra info.
     General,
     /// Forced failure via `AlwaysFalse`
     AlwaysFalse,
-    /// The attributes did not match expected:{expected:?} actual:{actual:?}
-    AttributeMismatch {
+    /// [`Attributes`] verification data
+    Attributes {
```

Comment:
Up for suggestions on these renames.  
In general I'm not fond of "metadata" as it's so generic it can often be ambiguous.

I'm fine with the variant names matching the underlying types, but not sure what others think.

Reply:
refactored, comment no longer relevant